ui/src/data-services/models/job-details.ts

```diff
@@ -0,0 +1,52 @@
+import { Job, ServerJob } from './job'
+
+export type ServerJobDetails = ServerJob & any // TODO: Update this type
+
+export class JobDetails extends Job {
+  public constructor(job: ServerJobDetails) {
+    super(job)
+  }
+
+  get inputLabel(): string {
+    return this._job.config.input.name
+  }
+
+  get inputValue(): string | number {
+    return this._job.config.input.size
+  }
+
+  get stages(): { key: string }[] {
+    return this._job.config.stages
+  }
+
+  getStageInfo(key: string) {
+    const stage = this._job.config.stages.find(
+      (stage: any) => stage.key === key
+    )
+
+    const progress = this._job.progress.stages.find(
+      (stage: any) => stage.key === key
+    )
+
+    if (!stage || !progress) {
+      return undefined
+    }
+
+    const name = stage.name
+    const status = this.getStatus(progress.status)
+    const statusLabel = this.getStatusLabel(status)
```

Comment:
This status label is basic, should probably include info about time remaining, but I was not fully sure about the logic here, so skipped for now. This is the place to tweak if needed :)

<img width="619" alt="Screenshot 2023-07-07 at 13 02 41" src="https://github.com/RolnickLab/ami-platform/assets/11680517/b31ccc6c-08d9-4d72-af18-92ee2d048ac1">

Reply:
I added a new "status label" to the stages progress. So `status` is the ENUM key with the status value. `progress` is the float value of the progress (from 0-1) and `status_label` is a formatted text string.

<img width="427" alt="image" src="https://github.com/RolnickLab/ami-platform/assets/158175/b4c3c22c-0b36-4ef6-8652-100cd426efe8">